src/libraries/Microsoft.Extensions.Primitives/tests/StringSegmentTest.cs

```diff
@@ -1268,9 +1268,12 @@ public void TrimEnd_RemovesTrailingWhitespaces(string value, int start, int leng
                     { null, null, StringComparison.InvariantCultureIgnoreCase, 0 },
                     { string.Empty, null, StringComparison.Ordinal, 1 },
                     { string.Empty, null, StringComparison.InvariantCultureIgnoreCase, 1 },
-                    { "x\u00E9y", "xE\u0301y", StringComparison.InvariantCulture, -1 }, // linguistic: lowercase sorts before uppercase
-                    { "x\u00E9y", "xE\u0301y", StringComparison.InvariantCultureIgnoreCase, 0 }, // equal (linguistic, one is normalized)
-                    { "Hello", "HELLO", StringComparison.InvariantCulture, -1 }, // linguistic: lowercase sorts before uppercase
+                    { "x\u00E9y", "xE\u0301y", StringComparison.InvariantCulture, 
+                    PlatformDetection.IsInvariantGlobalization && (PlatformDetection.IsiOS || PlatformDetection.IstvOS) ? 1 : -1 }, // linguistic: lowercase sorts before uppercase
```

Comment:
remove the `&& (PlatformDetection.IsiOS || PlatformDetection.IstvOS)`. having the `PlatformDetection.IsInvariantGlobalization` is enough :)

Reply:
Is to going to be a common case for all platforms, i.e. it's not apple-OSes specific?

Reply:
Removed iOS/tvOS-specific platform detection